src/tests/libecalc/presentation/yaml/test_venting_emitter.py

```diff
@@ -134,9 +129,22 @@ def test_venting_emitter_oil_volume(variables_map):
     }
     emissions_ch4 = emission_result["ch4"]
 
-    regularity_evaluated = float(
-        Expression.evaluate(regularity[datetime(1900, 1, 1)], fill_length=1, variables=variables_map.variables)
-    )
+    try:
+        regularity_array = Expression.evaluate(
+            regularity[datetime(1900, 1, 1)], fill_length=1, variables=variables_map.variables
+        )
+
+        if regularity_array.size == 0:
+            raise ValueError("Regularity array is empty, cannot proceed with test.")
+
+        regularity_evaluated = float(regularity_array[0])
+
+    except IndexError as e:
+        raise IndexError("Failed to evaluate regularity: array index out of range.") from e
+
+    except Exception as e:
+        raise RuntimeError("An unexpected error occurred during regularity evaluation in the test.") from e
+
     expected_result = [oil_value * regularity_evaluated * emission_factor / 1000 for oil_value in oil_values()]
```

Comment:
expected_result = [oil_value * expected_regularity * emission_factor / 1000 for oil_value in oil_values()]

Reply:
I have made some changes. Take a look if seems ok with the use of assert etc.

Reply:
It's ok, the test will fail if we break venting emitters. But I still think we could make it simpler by just using expected_regularity in expected_result. 

As far as I can see the try except basically ends up testing expressions.